### PR TITLE
Carpool v2: policy service migration

### LINKED ISSUE #2475: Support du carpool v2 sur le service policy
- ajout de la colonne "operator_journey_id"
- utilisation des données issues de carpool v2 en lieu et place de policy.trip
- continuer à utiliser le carpoolv1._id pour rétrocompat
-

### PR DESCRIPTION
**Pull request by nmrgt**
fix #2475 

## Summary by CodeRabbit

- **New Features**
  - Enhanced carpool policy incentives and data handling for improved service operations.
  - Updated carpool data structures with detailed identity and payment information.
  - Implemented validation patterns for identity keys to ensure data integrity.

- **Bug Fixes**
  - Adjusted carpool data seeding methods to meet new data requirements.

- **Refactor**
  - Migrated to a new version of carpool policies with additional database fields and indexes for better performance and uniqueness.

- **Documentation**
  - Updated key references and added new fields in policy engine helpers to align with the latest data handling changes.

### CODE REVIEW

File: api/src/pdc/services/policy/providers/TripRepositoryProvider.ts [hunk 1]
```diff
@@ -7,48 +7,91 @@ import { CarpoolInterface, PolicyInterface, TripRepositoryProviderInterfaceResol
...
-        FROM ${this.table} t
+          oc._id as _id,
+          oo.operator_uuid,
+          cc.operator_trip_id,
+          cc.operator_id,
+          cc.operator_journey_id,
+          cc.operator_class,
+          cc.passenger_contribution,
+          cc.passenger_identity_key,
+          (CASE WHEN cc.passenger_travelpass_user_id IS NULL THEN false ELSE true END) as passenger_has_travel_pass,
```
Comment (jonathanfallon) [inline api/src/pdc/services/policy/providers/TripRepositoryProvider.ts:40]: `cc.passenger_travelpass_user_id IS NOT NULL` suffit et est plus lisible

File: api/src/pdc/services/policy/providers/TripRepositoryProvider.ts [hunk 2]
```diff
@@ -7,48 +7,91 @@ import { CarpoolInterface, PolicyInterface, TripRepositoryProviderInterfaceResol
...
+          cc.passenger_over_18 as passenger_is_over_18,
+          cc.passenger_seats as seats,
+          cc.driver_revenue,
+          cc.driver_identity_key,
+          (CASE WHEN cc.driver_travelpass_user_id IS NULL THEN false ELSE true END) as driver_has_travel_pass,
```
Comment (jonathanfallon) [inline api/src/pdc/services/policy/providers/TripRepositoryProvider.ts:45]: cf. au dessus

File: api/src/pdc/services/policy/providers/TripRepositoryProvider.ts [hunk 3]
```diff
@@ -7,48 +7,91 @@ import { CarpoolInterface, PolicyInterface, TripRepositoryProviderInterfaceResol
...
+        }
         WHERE
-          (t.start_geo_code = ANY($1::varchar[]) OR t.end_geo_code = ANY($1::varchar[]))
-          AND t.datetime >= $2::timestamp
-          AND t.datetime < $3::timestamp
-          AND t.carpool_status = 'ok'
-        ORDER BY t.datetime ASC
+          cc.datetime >= $2::timestamp AND
+          cc.datetime < $3::timestamp AND
+          (co.start_geo_code = ANY($1::varchar[]) OR co.end_geo_code = ANY($1::varchar[]))
```
Comment (jonathanfallon) [inline api/src/pdc/services/policy/providers/TripRepositoryProvider.ts]: mettre les `AND` en début de ligne est + lisible

**Review by jonathanfallon — CHANGES_REQUESTED** [review]

File: api/src/pdc/services/policy/engine/helpers/limits.spec.ts [hunk 4]
```diff
@@ -214,7 +214,7 @@ test('should watch and apply for custom data', async (t) => {
   t.deepEqual(ctxStateless.meta.export(), [
     {
       uuid: 'uuid',
-      key: `max_passenger_restriction.${carpool.trip_id}.day.15-0-2019`,
+      key: `max_passenger_restriction.${carpool.operator_trip_id}.day.15-0-2019`,
```
[the remark this answers is not in the document]
Comment (nmrgt) [reply api/src/pdc/services/policy/engine/helpers/limits.spec.ts:219]: C'est un macro varchar donc il est sanatized. En revanche, j'ajoute l'operator_id pour éviter les conflits (improbable mais potentiel).

File: api/src/pdc/services/policy/engine/helpers/limits.spec.ts [hunk 5]
```diff
@@ -57,7 +57,7 @@ test('should watchForPersonMaxAmountByMonth', async (t) => {
   t.deepEqual(ctx.meta.export(), [
     {
       uuid: '1',
-      key: `max_amount_restriction.${LimitTargetEnum.Passenger}-${ctx.carpool.passenger_identity_uuid}.month.0-2019`,
+      key: `max_amount_restriction.${LimitTargetEnum.Passenger}-${ctx.carpool.passenger_identity_key}.month.0-2019`,
```
[the remark this answers is not in the document]
Comment (nmrgt) [reply api/src/pdc/services/policy/engine/helpers/limits.spec.ts:60]: Là, on est sur quelque chose de + normé c'est déjà un hash.

[the remark this answers is not in the document]
Comment (nmrgt) [reply api/src/pdc/services/policy/engine/helpers/limits.spec.ts:60]: j'ai mis à jour la validation du identity_key

File: api/src/pdc/services/policy/providers/TripRepositoryProvider.ts (hunk 2 above)
Comment (nmrgt) [reply api/src/pdc/services/policy/providers/TripRepositoryProvider.ts:45]: Yes

**Review by jonathanfallon — APPROVED** [review]

**Pull request merged**